system/lib/wasmfs/wasmfs.cpp

```diff
@@ -135,7 +135,8 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
 
   std::vector<std::string> pathParts;
 
-  auto newPathName = (char*)pathname;
+  char newPathName[strlen((char*)pathname)];
```

Comment:
Can't forget the space for the nul!
```suggestion
  char newPathName[strlen((char*)pathname) + 1];
```